### PR TITLE
DEV-27221: User API pagination update

### PR DESCRIPTION
**Pull request by george-haddad**
- Add note about `prev` and `next` being omitted when those pages do not exist
- Add 400 response error when `per_page` exceeds max of 500 (or is not between 1 and 500)

### CODE REVIEW

**Comment by tkrisztian95** [comment]
I think this should target the `develop` branch instead.

**Comment by george-haddad** [comment]
> I think this should target the `develop` branch instead.

Yeap, I think I just created this PR and it took `master` as the default.

File: apiary.apib [hunk 1]
```diff
@@ -1363,6 +1363,9 @@ within the `links` object:
 | `first`  | The URL to get the first page.    |
 | `last`   | The URL to get the last page.     |
 
+The `next` and `prev` link will be omitted when those pages do not exist. For example when
+viewing page 1 there `prev` link will be omitted because there is no resource to page 0.
+
```
Comment (rbarbey) [inline apiary.apib:1368]: I think we should also mention that the same applies to the next and prev page headers.

**Review by rbarbey — CHANGES_REQUESTED** [review]
I think there's one addition needed (for the headers) but otherwise it looks good. Also thank you for removing the draft section at the bottom!

File: apiary.apib (hunk 1 above)
Comment (george-haddad) [reply apiary.apib:1368]: Ah yes, true. I'll add that.

**Review by rbarbey — APPROVED** [review]
Thank you for the addition!

**Pull request merged**